vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java

```diff
@@ -442,22 +453,48 @@ public Result<Void, InvalidSpreadsheetRow> areInputsValid() {
       }
 
       // mandatory not filled in --> invalid
-      if (mandatoryInputs.stream().anyMatch(String::isBlank)) {
-        return Result.fromError(new InvalidSpreadsheetRow(
-            SpreadsheetInvalidationReason.MISSING_INPUT, rowId));
+      for(int colId : mandatoryInputCols) {
+        Cell cell = row.getCell(colId);
+        if(SpreadsheetMethods.cellToStringOrNull(cell).isBlank()) {
+          invalidCells.add(cell);
+        } else {
+          validCells.add(cell);
```

Comment:
why is it important to record the valid cells?

Reply:
The valid cells are recorded so the red highlighting can be removed when a user presses the registration button (and the cell was highlighted for missing data before). Unfortunately there is no possibility to remove it directly when inputting data, from what I can tell.